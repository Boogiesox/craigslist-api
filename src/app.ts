import express from 'express';
import { setupExpress } from './config/express';
import { handleMissing, handleErrors } from './middleware';
import { setupRoutesV1 } from './config/routes';

const app = express();

setupExpress(app);
setupRoutesV1(app);

app.use(handleMissing);
app.use(handleErrors);

export default app;
