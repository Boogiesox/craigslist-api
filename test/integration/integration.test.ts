/* eslint-disable @typescript-eslint/no-explicit-any */

jest.mock('nodemailer');
import request from 'supertest';

const mockedPutObject = jest.fn();
const mockedGetSignedUrl = jest.fn();
jest.mock('aws-sdk/clients/s3', (): any => {
    return class S3 {
        public putObject(params: any): any {
            mockedPutObject(params);
            return {
                /* eslint-disable @typescript-eslint/no-empty-function */
                promise: async (): Promise<void> => { }
            };
        }
        public getSignedUrl(operation: string, params: any): string {
            mockedGetSignedUrl(operation, params);
            return 'https://dummy.url.com';
        }
    };
});

import app from 'app';
describe('API V1', () => {
    describe('/hello', () => {
        describe('GET /', () => {
            it('should return 200 OK', async () => {
                const { body } = await request(app)
                    .get('/v1/hello')
                    .expect(200);
                expect(body).toMatchSnapshot();
            });
        });
    });
});
