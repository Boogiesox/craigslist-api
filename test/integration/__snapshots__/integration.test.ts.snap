// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API V1 /hello GET / should return 200 OK 1`] = `
Object {
  "success": true,
}
`;

exports[`API V1 /upload POST / should return 201, the document added to storage and call s3.putObject, s3.getSignedUrl - one file 1`] = `
Object {
  "data": Array [
    Object {
      "key": "GENERIC_UPLOAD_USER_ID/3eaf17869bb51bf27bd7c91bc9853973",
      "url": "https://dummy.url.com",
    },
  ],
}
`;
